clients/drcachesim/tools/filter/record_filter.cpp

```diff
@@ -603,8 +603,16 @@ record_filter_t::process_chunk_encodings(per_shard_t *per_shard, trace_entry_t &
             VPRINT(this, 3,
                    "output new-chunk encoding chunk=%" PRIu64 " ref=%" PRIu64 "\n",
                    per_shard->chunk_ordinal, per_shard->cur_refs);
-            if (!write_trace_entries(per_shard,
-                                     per_shard->per_input->pc2encoding[entry.addr])) {
+            // Sanity check that the encoding size is correct.
+            const auto &enc = per_shard->per_input->pc2encoding[entry.addr];
+            size_t enc_sz = 0;
+            for (const auto &record : enc)
+                enc_sz += record.size;
```

Comment:
We could figure this out  without looping: `(enc.size() - 1) * enc[0].size + enc[enc.size()-1].size`

Reply:
OK, though I'm not sure it's more readable?  The loop has just 1 or 2 iters nearly all the time.

Reply:
Fine either way.

Reply:
Added a comment.